### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/borg/actions/runs/3903581854#summary-10608236947

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/borg/actions/runs/3903582610#summary-10608239013

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -5,14 +5,14 @@ Thank you for reporting an issue.
  - Borgbackup documentation: http://borgbackup.readthedocs.io/en/stable/index.html
  - FAQ: https://borgbackup.readthedocs.io/en/stable/faq.html
  and
- - open issues in Github tracker: https://github.com/borgbackup/borg/issues
+ - open issues in GitHub tracker: https://github.com/borgbackup/borg/issues
```
Comment (jsoref) [inline .github/ISSUE_TEMPLATE.md:8]: Brand

File: docs/changes.rst [hunk 2]
```diff
@@ -333,7 +333,7 @@ Changes:
     - compression: use the 2 bytes for type and level, #6698
     - use version 2 for new archives
     - convert timestamps int/bigint -> msgpack.Timestamp, see #2323
-    - all hardlinks have chunks, maybe chunks_healty, hlid
+    - all hardlinks have chunks, maybe chunks_healthy, hlid
```
Comment (jsoref) [inline docs/changes.rst:336]: Some projects don't like changing changelogs, happy to drop this or anything else...

File: docs/faq.rst [hunk 3]
```diff
@@ -1156,7 +1156,7 @@ full when Borg starts a write command like `borg create`, it will abort
 immediately and the repository will stay as-is.
 
 If you run a backup that stops due to a disk running full, Borg will roll back,
-delete the new new segment file and thus freeing disk space automatically. There
+delete the new segment file and thus freeing disk space automatically. There
```
Comment (jsoref) [inline docs/faq.rst:1159]: I can't tell if this doubling is intentional or accidental

File: docs/misc/asciinema/Vagrantfile [hunk 4]
```diff
@@ -15,7 +15,7 @@ Vagrant.configure("2") do |config|
   SHELL
   config.vm.provision "record basic usage", type: "shell", inline: <<-SHELL
     # `rm` below allows quick re-exec via:
-    # vagrant vagrant provision --provision-with "record basic usage"
+    # vagrant provision --provision-with "record basic usage"
```
Comment (jsoref) [inline docs/misc/asciinema/Vagrantfile:18]: I'm _pretty_ sure this doubling is an error.

File: scripts/shell_completions/zsh/_borg [hunk 5]
```diff
@@ -1504,7 +1504,7 @@ _borg_compression() {
     'lz4:very high speed, very low compression'
   )
   local -a havelvl=(
-    'zstd:("zstandart")'
+    'zstd:("zstandard")'
```
Comment (jsoref) [inline scripts/shell_completions/zsh/_borg:1507]: I hope this is a typo...

**Review by jsoref — COMMENTED** [review]
Some corrections automatically suggested by Google Sheets. All fault mine

File: docs/changes.rst (hunk 2 above)
Comment (ThomasWaldmann) [reply docs/changes.rst:336]: changing is fine if it is correcting an error.

File: docs/faq.rst (hunk 3 above)
Comment (ThomasWaldmann) [reply docs/faq.rst:1159]: accidental

File: docs/misc/asciinema/Vagrantfile (hunk 4 above)
Comment (ThomasWaldmann) [reply docs/misc/asciinema/Vagrantfile:18]: yeah

File: src/borg/manifest.py [hunk 6]
```diff
@@ -148,7 +148,7 @@ class Manifest:
     class Operation(enum.Enum):
         # The comments here only roughly describe the scope of each feature. In the end, additions need to be
         # based on potential problems older clients could produce when accessing newer repositories and the
-        # tradeofs of locking version out or still allowing access. As all older versions and their exact
+        # trade offs of locking version out or still allowing access. As all older versions and their exact
```
Comment (ThomasWaldmann) [inline src/borg/manifest.py:151]: https://dict.leo.org/german-english/trade-off they say it is either `tradeoff` or `trade-off` (but not `trade off`).

File: src/borg/repository.py [hunk 7]
```diff
@@ -749,7 +749,7 @@ def check_free_space(self):
                     try:
                         compact_working_space += self.io.segment_size(segment) - free
                     except FileNotFoundError:
-                        # looks like self.compact is referring to a non-existent segment file, ignore it.
+                        # looks like self.compact is referring to a nonexistent segment file, ignore it.
```
Comment (ThomasWaldmann) [inline src/borg/repository.py:752]: https://dict.leo.org/german-english/non-existent seems to be BE vs AE.

I guess our docs are a wild mix between BE and AE...

File: src/borg/manifest.py (hunk 6 above)
Comment (jsoref) [reply src/borg/manifest.py]: I should have checked this one more closely. It didn't seem quite right to my mind, but it's too early in my morning.

Comment (jsoref) [reply src/borg/manifest.py]: ngrams says that `trade-offs` is winning:
https://books.google.com/ngrams/graph?content=tradeoffs%2Ctrade-offs&year_start=1800&year_end=2019&corpus=26&smoothing=3

https://trends.google.com/trends/explore?q=tradeoffs,trade-offs

Looks like the dashed form is preferred

**Comment by ThomasWaldmann** [comment]
About adding the spell checking action: guess that would be a good thing to avoid typos creeping into the docs again.

But, the amount of false positives (e.g. when checking the code) should be rather low or zero, so that the spell check is rather helping than being a permanent annoyance (e.g. if it would detect lots of typos in normal python code).

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**